File: snobot_sim_java/src/main/java/com/snobot/simulator/simulator_components/smart_sc/SmartScAnalogIn.java
package com.snobot.simulator.simulator_components.smart_sc;

import com.snobot.simulator.module_wrapper.ASensorWrapper;
import com.snobot.simulator.module_wrapper.interfaces.IAnalogInWrapper;

public class SmartScAnalogIn extends ASensorWrapper implements IAnalogInWrapper
{
    private double mVoltage;

    public SmartScAnalogIn(int aPort)
    {
        super("CAN Analog (" + (aPort - BaseCanSmartSpeedController.sCAN_SC_OFFSET) + ")");
    }

    @Override
    public void setVoltage(double aVoltage)
    {
        mVoltage = aVoltage;
    }

    @Override
    public double getVoltage()
    {
        return mVoltage;
    }

}

File: snobot_sim/src/main/native/cpp/SnobotSim/SimulatorComponents/SmartSC/SmartSCAnalogIn.cpp
#include "SnobotSim/SimulatorComponents/SmartSC/SmartSCAnalogIn.h"

#include "SnobotSim/SimulatorComponents/SmartSC/CanIdOffset.h"

const std::string SmartSCAnalogIn::TYPE = "com.snobot.simulator.simulator_components.smart_sc.SmartScAnalogIn";

SmartSCAnalogIn::SmartSCAnalogIn(int aPort) :
        AModuleWrapper("CAN Analog (" + std::to_string(aPort - CAN_SC_OFFSET) + ")")
{
}

SmartSCAnalogIn::~SmartSCAnalogIn()
{
}

void SmartSCAnalogIn::SetVoltage(double aVoltage)
{
    mVoltage = aVoltage;
}

double SmartSCAnalogIn::GetVoltage()
{
    return mVoltage;
}
